bot/exts/events/advent_of_code/_cog.py

```diff
@@ -172,6 +175,72 @@ async def join_leaderboard(self, ctx: commands.Context) -> None:
         else:
             await ctx.message.add_reaction(Emojis.envelope)
 
+    @in_month(Month.NOVEMBER, Month.DECEMBER)
+    @adventofcode_group.command(
+        name="link",
+        aliases=("connect",),
+        brief="Tie your Discord account with your Advent of Code name."
+    )
+    @whitelist_override(channels=AOC_WHITELIST)
+    async def aoc_link_account(self, ctx: commands.Context, aoc_name: str = None) -> None:
+        """
+        Link your Discord Account to your Advent of Code name.
+
+        Stored in a Redis Cache, Discord ID: Advent of Code Name
+        """
+        cache_items = await self.account_links.items()
+
+        # A short circuit in case the cache is empty
+        if len(cache_items) == 0 and aoc_name:
+            log.info(f"{ctx.author} ({ctx.author.id}) is now linked to {aoc_name}.")
+            await self.account_links.set(ctx.author.id, aoc_name)
+            await ctx.reply(f"You have linked your Discord ID to {aoc_name}.")
+            return
+        elif len(cache_items) == 0:
+            await ctx.reply(
+                "You have not linked an Advent of Code account."
+                "Please re-run the command with one specified."
+            )
+            return
```

Comment:
If the cache is empty, later code trying to pull items from the cache will fail. Additionally, this will help with future development. Did you test this locally?

Reply:
Yeah, it works

![Screenshot from 2021-12-03 07-34-49](https://user-images.githubusercontent.com/69356296/144532573-3b2b7a8f-4c31-4dfc-a7c3-1f5dd86fb144.png)